core/connmgr/manager.go

```diff
@@ -89,3 +93,9 @@ type TagInfo struct {
 	// Conns maps connection ids (such as remote multiaddr) to their creation time.
 	Conns map[string]time.Time
 }
+
+// GetterConnLimit provides access to a component's total connection limit.
+type GetterConnLimit interface {
+	// GetConnLimit returns the total connection limit of the implementing component.
```

Comment:
I'd prefer `GetConnLimiter` or `ConnLimitGetter` over the current one. 
Between the two slight preference for `GetConnLimiter` since it is inline with [effective-go](https://go.dev/doc/effective_go) recommendation:
>By convention, one-method interfaces are named by the method name plus an -er suffix or similar modification to construct an agent noun: Reader, Writer, Formatter, CloseNotifier etc.